skimage/future/graph/rag.py

```diff
@@ -264,6 +272,30 @@ def _add_node_silent(self, n):
         .. seealso:: :func:`networkx.Graph.add_node`."""
         super(RAG, self).add_node(n)
 
+    def nodes_iter(self, *args, **kwargs):
```

Comment:
I did but don't have an immediate answer... But, when in doubt, I think it's safer to keep @matthew-brett's changes. I'm inclined to think they are very much valid concerns.